sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -855,6 +855,15 @@ object ColumnPruning extends Rule[LogicalPlan] {
     case e @ Expand(_, _, child) if !child.outputSet.subsetOf(e.references) =>
       e.copy(child = prunedChild(child, e.references))
 
+    // Prunes the duplicate columns from child of UnaryNode, e.g.: Aggregate
```

Comment:
Hm, I think this doesn't work for now. Spark currently allows duplicated column names in some Dataset API cases too.

Reply:
OK. will close this pr.